src/cgame/rocket/rocket.cpp

```diff
@@ -525,99 +538,60 @@ Rml::String Rocket_QuakeToRML( const char *in, int parseFlags = 0 )
 		{
 			char c = *token.Begin();
 			const emoticonData_t *emoticon;
-			if ( c == '<' )
-			{
-				if ( span && !spanHasContent )
-				{
-					spanHasContent = true;
-					out.append( spanstr );
-				}
-				out.append( "&lt;" );
-			}
-			else if ( c == '>' )
-			{
-				if ( span && !spanHasContent )
-				{
-					spanHasContent = true;
-					out.append( spanstr );
-				}
-				out.append( "&gt;" );
-			}
-			else if ( c == '&' )
-			{
-				if ( span && !spanHasContent )
-				{
-					spanHasContent = true;
-					out.append( spanstr );
-				}
-				out.append( "&amp;" );
-			}
-			else if ( c == '\n' )
+
+			std::unordered_map<char, std::string> htmlEscapeStrings;
+			htmlEscapeStrings['<'] = "&lt;";
+			htmlEscapeStrings['>'] = "&gt;";
+			htmlEscapeStrings['&'] = "&amp;";
+			htmlEscapeStrings['"'] = "&quot;";
+			htmlEscapeStrings['\''] = "&apos;";
+			htmlEscapeStrings['\n'] = "<br />";
+
+			std::string htmlEscapedString = htmlEscapeStrings[ c ];
+
+			if ( htmlEscapedString != "" )
```

Comment:
use .empty()? or do

```
auto it = htmlEscapeStrings.find( c );
if ( htmlEscapeStrings != htmlEscapeStrings.end() )
{
    out.append( it.second );
}
else
{
    // handle emoticon
}

Reply:
I get:

```
error: ‘struct std::__detail::_Node_const_iterator<std::pair<const char,
  std::__cxx11::basic_string<char> >, false, false>’ has no member named ‘second’
```

Reply:
Also the compiler doesn't let me make an `unordered_map` with a `const char`.

Reply:
More precisely, in this situation the compiler does not allow me to make a `const unordered_map` with a `char` (but it also does not allow to make a an `unordered_map` with a `const char`).

I turned it into a static global I never modify anyway.

Reply:
Also now uses `.empty()`.

Reply:
I always get mixed up. Maybe `it->second`?